src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -549,14 +554,35 @@ void network::save(cldnn::BinaryOutputBuffer& ob) {
     size_t exec_order_size = _exec_order.size();
     ob << exec_order_size;
 
+    std::unordered_map<primitive_id, int> exec_order_num;
+    int i = -1;
     for (const auto& p_inst : _exec_order) {
+        exec_order_num[p_inst->id()] = i--;
+    }
+
+    std::vector<std::shared_ptr<primitive_inst>> insts_to_allocate(_exec_order.begin(), _exec_order.end());
+    std::sort(insts_to_allocate.begin(),
+              insts_to_allocate.end(),
+              [&exec_order_num](std::shared_ptr<primitive_inst> const& lhs, std::shared_ptr<primitive_inst> const& rhs) {
+                    long lhs_size = (lhs->mem_allocated()) ? lhs->get_output_layout().bytes_count() : exec_order_num[lhs->id()];
+                    long rhs_size = (rhs->mem_allocated()) ? rhs->get_output_layout().bytes_count() : exec_order_num[rhs->id()];
+                    return (lhs_size > rhs_size);
+              });
+
```

Comment:
Seems that it's sorting twice, once in the allocate_instance phase, and then here in save. 
Could we store the first sort result and then just use it here?

Reply:
They are different in that this sorting logic moves primitive_insts that are not need to allocate memory to the end of the list.
This is necessary to reuse memory in these primitive_insts.